lane.py

```diff
@@ -1201,8 +1197,22 @@ def _defer_unused_opds_entry(self, query, work_model=Work):
         else:
             return query.options(defer(work_model.simple_opds_entry))
 
-    def featured_works(self, size, use_materialized_works=True):
-        """Find a random sample of `size` featured books.
+    def sublane_samples(self, use_materialized_works=True):
+        """Generates a list of samples from each sublane for a groups feed"""
+
+        # This is a list rather than a dict because we want to
+        # preserve the ordering of the lanes.
+        works_and_lanes = []
+        for sublane in self.visible_sublanes:
+            works = sublane.featured_works(
+                use_materialized_works=use_materialized_works
+            )
+            for work in works:
+                works_and_lanes.append((work, sublane))
+        return works_and_lanes
+
+    def featured_works(self, use_materialized_works=True):
+        """Find a random sample of featured books.
 
         It's semi-okay for this to be slow, since it will only be run to
         create cached feeds.
```

Comment:
This is no longer true. It's true for the default implementation but not necessarily true for the subclass implementation.